webapp/src/assetEditor.tsx

```diff
@@ -16,11 +16,26 @@ function init() {
 }
 
 export class AssetEditor extends React.Component {
+    private editor: FieldEditorComponent<any>;
+
+    handleMessage = (msg: any)  => {
+        if (msg.data._fromVscode && msg.data.type === "update") {
```

Comment:
you should define interface that specify the message format.

Reply:
can data be null/undefined? use ?.

Reply:
why is there a underscore in ``_fromVscode``? it does not really follow any convention.

Reply:
![image](https://user-images.githubusercontent.com/15070078/87604104-42841c80-c6c6-11ea-941d-a6e2c0755871.png)
For the '_fromVscode', I just tried to follow what was written for simulator messages in the vscode extension. I'm not sure the reason for the underscore, maybe because it should always be set to true for messages from the extension?

Reply:
Yea the '_fromVscode' is the way it's specified in the simulator messages in the vscode extension so I just thought I should be consistent, but I can change it.

Reply:
we should specify the message format here in makecode for consumers. I would drop the ``_fromVscode`` and instead use something like ``"source": "vscode"``

Reply:
The message will always have a data attribute.